third_party/3/frozendict.pyi

```diff
@@ -0,0 +1,27 @@
+import collections
+from typing import Any, Dict, Generic, Iterable, Iterator, Mapping, Tuple, Type, TypeVar, overload
+
+_S = TypeVar("_S")
+_KT = TypeVar("_KT")
+_VT = TypeVar("_VT")
+
+class frozendict(Mapping[_KT, _VT], Generic[_KT, _VT]):
+
+    dict_cls: Type[Dict] = ...
+    @overload
+    def __init__(self, **kwargs: _VT) -> None: ...
+    @overload
+    def __init__(self, mapping: Mapping[_KT, _VT]) -> None: ...
+    @overload
+    def __init__(self, iterable: Iterable[Tuple[_KT, _VT]]) -> None: ...
+    def __getitem__(self, key: _KT) -> _VT: ...
+    def __contains__(self, key: object) -> bool: ...
+    def copy(self: _S, **add_or_replace) -> _S: ...
```

Comment:
```suggestion
    def copy(self: _S, **add_or_replace: _VT) -> _S: ...
```

Ideally, we'd be able to restrict `_S` to `str` keys here, but I have no good idea how to do that.

Reply:
suggested change applied, thanks!
it's possible that the frozendict has mixed key types, though it's true most of the time it's used with str keys only.